components/plurals/build.rs

```diff
@@ -0,0 +1,13 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+fn main() {
```

Comment:
thought: really not a fan of introducing more build scripts; they're a pain for non-Cargo users, and people in general do not like them.

Would prefer the crate based solution

Reply:
In the issue we decided 

> globaldata can be replaced with a custom baked rust source using an env var.

There's no way to do defaulting env vars without build scripts.

Reply:
How does a separate crate solve this? It would only move the problem, you'd still have a build script there. It would also introduce another problems around semver, we'd need to use a `=` version range, but that's also frowned upon.

Reply:
Hmm. I was under the impression we were still considering doing crates. I think crate replacement works out far easier.

I would ideally like a non-build-script ICU4X to still _work_ with the data feature. To do this you'd have to use a `cfg` that's detected from the env var. Basically:

 - `ICU4X_PLURALS_REPLACE_DATA = ...` is the input
 - Build script detects it and sets `--cfg data-replacement` (see CPTBuilder's `icu4c_enable_renaming` for how this works)
 - Code uses `cfg(data-replacement)` to replace the relative path include with `include!(env!("ICU4X_PLURALS_REPLACE_DATA"))`

This means code using build scripts needs to set an env var _and_ a CFG, but that's not an undue burden IMO.

(If we're not doing crates, we don't really need macros anyway, right?)

Reply:
Aha, I see.

Reply:
I hope we do not risk data duplication there. *might* be easier on the compiler to set up a fork-by-key provider for that

Reply:
We *could* actually do this without a build script if we had a `builtin-data` and a `custom-data` feature instead of feature + automatic cfg.

Reply:
Yeah dueling features are annoying

Reply:
Now thinking of it, it's really not correct that `_with_any_provider` and `_with_buffer_provider` call `_unstable` unconditionally; we see this problem in icu_locid_transform. It is more correct to call the current constructor type all the way down the stack, only converting to a DataProvider at the last minute when calling actual `load` functions. So, all four constructors should follow the same pattern of calling the same constructor of child classes and combining the results with `load` calls on the current crate's data structs.

Reply:
We seem to agree on the *Custom globaldata path* issue, and *Macros in crates* is not a discussion we need to have on this PR. I'm not calling macros from other crates.